core/src/main/java/com/godaddy/commerce/core/http/HttpVerbWrapper.java

```diff
@@ -504,6 +511,17 @@ private <T> HttpVerbResult<T> callHttp(Class<? extends T> responseClaz, String u
     }
   }
 
+  private void addXAppKeyHeader(HttpRequestBase request) {
+    AppConfig appConfig = AppConfigProvider.appConfig();
+
+    String hostName = Optional.ofNullable(appConfig)
```

Comment:
> [<img alt="prajasekaran" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/prajasekaran) **Authored by [prajasekaran](https://github.secureserver.net/prajasekaran)**
_<time datetime="2021-11-05T18:13:34Z" title="Friday, November 5th 2021, 7:13:34 pm +01:00">Nov 5, 2021</time>_

---


```suggestion
    String xAppKey = Optional.ofNullable(AppConfigProvider.appConfig())
```

Reply:
> [<img alt="esarkisian" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/esarkisian) **Authored by [esarkisian](https://github.secureserver.net/esarkisian)**
_<time datetime="2021-11-05T18:20:00Z" title="Friday, November 5th 2021, 7:20:00 pm +01:00">Nov 5, 2021</time>_

---


fixed